src/components/CalendarMonthGrid.jsx

```diff
@@ -146,9 +148,11 @@ class CalendarMonthGrid extends React.Component {
       const withoutTransitionMonths = orientation === VERTICAL_SCROLLABLE;
       newMonths = getMonths(initialMonth, numberOfMonths, withoutTransitionMonths);
     }
-    
-    var momentLocale = moment.locale();
-    newMonths = newMonths.map(m => m.locale(momentLocale));
+
+    if (this.locale !== moment.locale()) {
```

Comment:
Let’s cache the call to locale here, so we don’t have to do it twice

Reply:
@krtcom You still here? I'd love to have this updated, and I'm not exactly sure if I can add to your pull request, but I believe all you need to do is this.

var momentLocale = moment.locale();
if (this.locale !== momentLocale) {
 +      this.locale = momentLocale;
 +      newMonths = newMonths.map(m => m.locale(this.locale));
 +    }

Reply:
@humphries40 if you want to cherry-pick this onto another branch and open a PR with the fix, we would def roll that in!

Reply:
Hi there, sorry I was busy with something else, and postponing this one but I can find some time to change it this morning.